In [10]:
library(tidyverse)
library(repr)
library(infer)
#library(tidymodels)
library(cowplot)
options(repr.matrix.max.rows = 6)
source('cleanup.R')

## Research Question

Can a player’s total cumulative play-time (`total_minutes`)
and average session length (`mean_session`)
predict whether they subscribe to the game’s newsletter?

**Response variable:** `newsletter_subscribed`  
(0 = no, 1 = yes)  
**Predictors:** `total_minutes`, `mean_session`

In [6]:
players <- read_csv("Data/players.csv")
sessions <- read_csv("Data/sessions.csv")

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


In [14]:
print("Players Data")
head(players)
tail(players)
dim(players)

print("Sessions Data")
head(sessions)
tail(sessions)
dim(sessions)

[1] "Players Data"


experience subscribe
1 Pro         TRUE    
2 Veteran     TRUE    
3 Veteran    FALSE    
4 Amateur     TRUE    
5 Regular     TRUE    
6 Amateur     TRUE    
  hashedEmail                                                      played_hours
1 f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d 30.3        
2 f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9  3.8        
3 b674dd7ee0d24096d1c019615ce4d12b20fcbff12d79d3c5a9d2118eb7ccbb28  0.0        
4 23fe711e0e3b77f1da7aa221ab1192afe21648d47d2b4fa7a5a659ff443a0eb5  0.7        
5 7dc01f10bf20671ecfccdac23812b1b415acd42c2147cb0af4d48fcce2420f3e  0.1        
6 f58aad5996a435f16b0284a3b267f973f9af99e7a89bee0430055a44fa92f977  0.0        
  name      gender Age
1 Morgan    Male    9 
2 Christian Male   17 
3 Blake     Male   17 
4 Flora     Female 21 
5 Kylie     Male   21 
6 Adrian    Female 17

experience subscribe
1 Amateur     TRUE    
2 Amateur     TRUE    
3 Veteran    FALSE    
4 Amateur    FALSE    
5 Amateur    FALSE    
6 Pro         TRUE    
  hashedEmail                                                      played_hours
1 644fe4200c19a73768b3fa598afcbd0948f7557925b7f17166285da23af31cc6 0.0         
2 b6e9e593b9ec51c5e335457341c324c34a2239531e1890b93ca52ac1dc76b08f 0.0         
3 71453e425f07d10da4fa2b349c83e73ccdf0fb3312f778b35c5802c3292c87bd 0.3         
4 d572f391d452b76ea2d7e5e53a3d38bfd7499c7399db299bd4fedb06a46ad5bb 0.0         
5 f19e136ddde68f365afc860c725ccff54307dedd13968e896a9f890c40aea436 2.3         
6 d9473710057f7d42f36570f0be83817a4eea614029ff90cf50d8889cdd729d11 0.2         
  name   gender            Age
1 Rhys   Male              20 
2 Bailey Female            17 
3 Pascal Male              22 
4 Dylan  Prefer not to say 17 
5 Harlow Male              17 
6 Ahmed  Other             NA

[1] 196   7

[1] "Sessions Data"


hashedEmail                                                     
1 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
2 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
3 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
4 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
5 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
6 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
  start_time       end_time         original_start_time original_end_time
1 30/06/2024 18:12 30/06/2024 18:24 1.71977e+12         1.71977e+12      
2 17/06/2024 23:33 17/06/2024 23:46 1.71867e+12         1.71867e+12      
3 25/07/2024 17:34 25/07/2024 17:57 1.72193e+12         1.72193e+12      
4 25/07/2024 03:22 25/07/2024 03:58 1.72188e+12         1.72188e+12      
5 25/05/2024 16:01 25/05/2024 16:12 1.71665e+12         1.71665e+12      
6 23/06/2024 15:08 23/06/2024 17:10 1.71916e+12         1.71916e+12

hashedEmail                                                     
1 b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
2 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
3 7a4686586d290c67179275c7c3dfb4ea02f4d317d9ee0e2cee98baa27877a875
4 fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
5 fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
6 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
  start_time       end_time         original_start_time original_end_time
1 28/08/2024 23:31 29/08/2024 00:15 1.72489e+12         1.72489e+12      
2 10/05/2024 23:01 10/05/2024 23:07 1.71538e+12         1.71538e+12      
3 01/07/2024 04:08 01/07/2024 04:19 1.71981e+12         1.71981e+12      
4 28/07/2024 15:36 28/07/2024 15:57 1.72218e+12         1.72218e+12      
5 25/07/2024 06:15 25/07/2024 06:22 1.72189e+12         1.72189e+12      
6 20/05/2024 02:26 20/05/2024 02:45 1.71617e+12         1.71617e+12

[1] 1535    5

**What is the dataset about?**  
This data comes from a UBC Minecraft research server.  
- `players.csv` has one row per player (demographics, total play time, newsletter flag).  
- `sessions.csv` logs each play session’s start/end timestamps so we can derive session lengths.

**How many variables?**  
- `players.csv`: 7 columns  
- `sessions.csv`: 5 columns

**How many observations?**  
- `players.csv`: 196 rows  
- `sessions.csv`: 1535 rows

In [8]:
source('cleanup.R')